### ISSUE TITLE
4.21: Checking Controller status in packaged build results in "VRInputError_WrongType" input errors

### ISSUE THREAD

**Issue by smapty** [ValveSoftware/steamvr_unreal_plugin#49]
In our game, we check to see the number of motion controllers that are tracked to see how we should adjust player input. When we run FSteamVRInputDevice::GetControllerTrackingStatus, the VRInput()->GetPoseActionDataForNextFrame functions return a VRInputError_WrongType error.
I originally thought that the editor wasn't experiencing this issue, but it did when I first ran it. Eventually it was working and only packaged builds exhibited the issue.

**Comment by 1runeberg** [comment]
Hi @smapty - does that mean, you're not getting tracking on your packaged builds? or you are just seeing log entries on a dev/debug build?

**Comment by smapty** [comment]
Hey @1runeberg,

I did some more investigation and found the following. Hopefully, this gets to what you were wondering.

I was having issues using the GetPoseActionDataForNextFrame call and decided to use the GetPoseActionDataRelativeToNow instead. When I initially did that, the status came back correctly as EVRInputError::VRInputError_None. The problem was that the motion controller poses weren't being updated after that. I went back again and modified FSteamVRInputDevice::GetControllerTrackingStatus to use the GlobalPredictedSecondsFromNow variable to choose between the NextFrame or RelativeToNow functions and that worked for both fixing the tracking status and getting the pose.

tldr; I wasn't getting tracking or controller status'. After modifying the plugin to use PoseActionDataRelativeToNow, both the tracking status and controller tracking is working.
[SteamVRInputDevice.zip](https://github.com/ValveSoftware/steamvr_unreal_plugin/files/3335991/SteamVRInputDevice.zip)

**Comment by 1runeberg** [comment]
Hi @smapty - am a bit confused, the current published version of the plugin already does this? are you perhaps using an older version?

**Comment by smapty** [comment]
This is a published version with the filename...
```steamvr_unreal_plugin-1.0b_4.21.zip```
Let me know if there's a newer version we should be using.

I saw that the module would switch between NextFrame and RelativeToNow for FSteamVRInputDevice::GetControllerOrientationAndPosition based on GlobalPredictedSecondsFromNow but, FSteamVRInputDevice::GetControllerTrackingStatus didn't. I added the same functionality to that function and then had to modify the GlobalPredictedSecondsFromNow variable to prevent it from using the NextFrame pose function.

**Comment by 1runeberg** [comment]
ah.. gotcha. have to look at more closely why thats happening, as NextFrame & RelativeToNow with a zero input should be the same call in the steamvr runtime. 

what value are you putting in GlobalPredictedSecondsFromNow?

**Comment by 1runeberg** [comment]
Addressed in develop branches.
